docs/advanced/full-node.md

```diff
@@ -0,0 +1,63 @@
+- [Installing dcrd](https://stakey.club/en/installing-dcrd/): A more advanced guide to installing a full node (Mac or Debian) that also covers manual installation (from binaries or source) and configuring dcrd to run on Tor for extra security.
+
+## Steps
+
+Below are the basic high-level steps to install a full node on any hardware and OS. These instructions use [dcrinstall](../wallets/cli/cli-installation.md), the automatic installer and upgrader software. In addition to installing `dcrd`, it will also install `dcrwallet` (which allows you to create wallets) and `dcrctl` (which allows you to control `dcrwallet` and `dcrd` from the command line). You can also just install `dcrd` on its own from [binaries or source files](https://github.com/decred/dcrd#installing-and-updating); a full node does not require a wallet unless you want to use your full node to validate and broadcast your own transactions.
+
+1. **Download and install dcrd**
+	- Follow the instructions for your OS in the [CLI Installation guide](https://docs.decred.org/wallets/cli/cli-installation/). The `dcrd` executable will be installed in a directory named `/decred` under your OS's home directory. 
+2. **Start `dcrd`**
+	- Navigate to the `/decred` directory and launch `dcrd`. See the [Operating System Differences](../wallets/cli/os-differences.md) page for OS-specific commands.
+	- `dcrd` will boot up, begin connecting to peers and downloading the full Decred blockchain.
+3. **(Optional) Create wallet**
+	- If you would like to use your full node to validate and broadcast your wallet's transactions, you'll need to create and set up a wallet. See the [dcrwallet Setup Guide](../wallets/cli/dcrwallet-setup.md) for instructions.
+4. **Enable incoming connections**
+	- `dcrd` will automatically begin downloading the blockchain and connecting to peers. However, if you want to help the network and have your node start validating transactions, you'll need to configure your internet router to allow inbound connections. 
+		1. Find your [Local IP address](https://www.howtogeek.com/236838/how-to-find-any-devices-ip-address-mac-address-and-other-network-connection-details/) and [External IP address](https://whatismyipaddress.com/).
+		2. Edit the settings of your router to open port 9108 (port forwarding). This [process](https://www.wikihow.com/Set-Up-Port-Forwarding-on-a-Router) depends on the type of router you have. 
+	- If running a firewall, you'll also need to configure it to allow connections on port 9108.
+5. **Wait for dcrd to sync**
+	- Check a Decred block explorer such as [dcrdata](https://dcrdata.decred.org/) and wait until the block height displayed matches the block height shown locally.
+5. **Restart dcrd**
+	- Restart `dcrd` and pass it your external IP address using the following flag `--externalip=[your.external.ip]`
```

Comment:
I think it makes more sense to set your external IP address **before** starting dcrd for the first time.

Nodes maintain a list of accessible/non-accessible peers, and coming online whilst not accessible may result in your node being added to the non-accessible lists, leading to peers ignoring your node in future.

@davecgh is my understanding correct here?

Reply:
Yes, but it won't matter with 1.5.0 anymore for most cases anyway, because it will now automatically discover your external IP address based on what the other peers think you are.

That said, it can still be overridden with the `externalip` flag.

Reply:
When will v1.5.0 be released, roughly? Unless it's going to happen soon, I'm inclined to just document it assuming v1.4.0. However, how else would one set the externalip? Only seeing how to pass externalip when restarting dcrd...would the user just restart it twice?

Reply:
We're aiming for the RC of 1.5.0 within 2 weeks.  Correct you can only set externalip on the command line.

Reply:
OK lets not get caught up on this. We can just go with the text as-is, and we can revisit when 1.5 comes out.